batchflow/tests/classification_metrics_test.py

```diff
@@ -0,0 +1,175 @@
+"""File contains various tests for classification metrics in batchflow"""
+# pylint: disable=import-error, no-name-in-module
+# pylint: disable=redefined-outer-name
+import pytest
+import numpy as np
+
+from batchflow.models.metrics import ClassificationMetrics as cm
+
+# tests for confusion matrix
+
+def test_confusion_matrix():
+    """Test on correctness of confusion matrix calculation"""
+    y_true, y_pred = np.array([1, 1, 0, 1, 0, 0]), np.array([0, 0, 1, 0, 0, 0])
+    conf_matrix = np.array([[2, 3], [1, 0]])
+    conf_matrix_calc = cm(y_true, y_pred, fmt='labels', num_classes=2)._confusion_matrix  #pylint:disable=protected-access
+    assert (conf_matrix_calc == conf_matrix).all()
+
+def test_confusion_matrix_multiclass():
+    """Test on correctness of confusion matrix calculation for multiclass"""
+    y_true, y_pred = np.array([2, 1]), np.array([[0.1, 0.1, 0.8], [0.1, 0.8, 0.1]])
+    conf_matrix = np.array([[0, 0, 0], [0, 1, 0], [0, 0, 1]])
+    conf_matrix_calc = cm(y_true, y_pred, fmt='proba', axis=1, num_classes=3)._confusion_matrix  #pylint:disable=protected-access
+    assert (conf_matrix_calc == conf_matrix).all()
+
+# tests on shapes
+
+@pytest.fixture(params=['accuracy', 'f1_score', 'true_positive_rate', 'false_positive_rate'])
+def metrics_name(request):
+    return request.param
```

Comment:
That fixture is used only as a container for parameters. You can instead use parametrization for class and put your test functions in it